### DISCUSSION TITLE
Lesson 6

### DISCUSSION THREAD

**Discussion by MoonChild362000** [Cyfrin/foundry-full-course-cu#818] [Q&A]
I am having a problem where i have already import my Simple Storage but it is said that my (SimpleStorage) to be unnamed and unassigned.

My issues
'''
Unnamed return variable can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the variable.
'''

Below are my code. 

'''solidity

// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {Script} from "../lib/forge-std/src/Script.sol";
import {SimpleStorage} from "/home/moonchild/src/SimpleStorage.sol";

contract DeploySimpleStorage is Script{
    function run() external returns (SimpleStorage) {
        vm.startBroadcast();
        SimpleStorage simpleStorage= new SimpleStorage();
        vm.stopBroadcast();
    }
}

'''

**Comment by elec7rik** [comment]
hi @MoonChild362000 , the function run() is supposed to return an instance of type SimpleStorage as defined in the function definition.
```function run() external returns (SimpleStorage)```
so i think you are missing the return statement there, something like:
```return simpleStorage;```
i hope this resolves the error.
